modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryBootstrap.java

```diff
@@ -0,0 +1,163 @@
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map.Entry;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.ApplicationContext;
+
+/**
+ * Specialization of {@link SimpleRegistryBootstrap which instead of registering the objects directly
+ * into a {@link Registry}, generates {@link BeanDefinition}s into a {@link BeanDefinitionRegistry} so
+ * that the Spring framework can create those objects when initialising an {@link ApplicationContext}}
+ *
+ * @since 3.7.0
+ */
+public class SpringRegistryBootstrap extends SimpleRegistryBootstrap implements BeanFactoryAware
+{
+
+    private BeanDefinitionRegistry beanDefinitionRegistry;
+
+    public SpringRegistryBootstrap()
+    {
+        super(new ClassPathRegistryBootstrapDiscoverer());
+    }
+
+    @Override
+    public void setBeanFactory(BeanFactory beanFactory) throws BeansException
+    {
+        checkArgument(beanFactory instanceof BeanDefinitionRegistry, "this bootstrap class only accepts BeanFactory instances which implement " + BeanDefinitionRegistry.class);
+        beanDefinitionRegistry = (BeanDefinitionRegistry) beanFactory;
+    }
+
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        super.initialise();
+        try
+        {
+            monopolizeRegistries();
+        }
+        catch (Exception e)
+        {
+            throw new InitialisationException(e, this);
+        }
+    }
+
+    @Override
+    protected void registerTransformers() throws MuleException
+    {
+        //no-op .. will happen on post processors
+    }
+
+    @Override
+    protected void doRegisterTransformer(String name, Class<?> returnClass, Class<? extends Transformer> transformerClass, String mime) throws Exception
+    {
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.rootBeanDefinition(transformerClass);
+
+        DataType returnType = null;
+
+        if (returnClass != null)
+        {
+            returnType = DataTypeFactory.create(returnClass, mime);
+            builder.addPropertyValue("returnDataType", returnType);
+        }
+
+        if (name == null)
+        {
+            //This will generate a default name for the transformer
+            //We then prefix the name to ensure there is less chance of conflict if the user registers
+            // the transformer with the same name
+            name = "_" + TransformerUtils.generateTransformerName(transformerClass, returnType);
+        }
+
+        builder.addPropertyValue("name", name);
+        doRegisterObject(name, builder);
+    }
+
+    private void monopolizeRegistries()
```

Comment:
Why is this here and not elsewhere?  What has this got to do with the bootstrap properties?